gadgets/snapshot_process/test/unit/snapshot_process_test.go

```diff
@@ -0,0 +1,165 @@
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, sleepPid int) *ExpectedSnapshotProcessEvent {
+					return &ExpectedSnapshotProcessEvent{
+						Comm:      "sleep",
+						Pid:       sleepPid,
+						Tid:       sleepPid,
+						Uid:       0,
+						Gid:       0,
+						ParentPid: info.Tid,
+						MountNsID: info.MountNsID,
+					}
+				})(t, info, fd, events)
+				return nil
+			},
+		},
+		"captures_no_events_with_no_matching_filter": {
+			runnerConfig: runnerConfig,
+			mntnsFilterMap: func(info *utilstest.RunnerInfo) *ebpf.Map {
+				mnts, _ := containerutils.GetNetNs(os.Getpid())
+				return utilstest.CreateMntNsFilterMap(t, mnts)
+			},
+			generateEvent: generateEvent,
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedSnapshotProcessEvent) error {
+				utilstest.ExpectNoEvent(t, info, fd, events)
+				return nil
+			},
+		},
+		"captures_events_with_matching_filter": {
+			mntnsFilterMap: func(info *utilstest.RunnerInfo) *ebpf.Map {
+				return utilstest.CreateMntNsFilterMap(t, info.MountNsID)
+
+			},
+			generateEvent: generateEvent,
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedSnapshotProcessEvent) error {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, sleepPid int) *ExpectedSnapshotProcessEvent {
+					return &ExpectedSnapshotProcessEvent{
+						Comm:      "sleep",
+						Pid:       sleepPid,
+						Tid:       sleepPid,
+						Uid:       0,
+						Gid:       0,
+						ParentPid: info.Tid,
+						MountNsID: info.MountNsID,
+					}
+				})(t, info, fd, events)
+				return nil
+			},
+		},
+		"no_threads_are_captured": {
+			runnerConfig:  runnerConfig,
+			generateEvent: generateEvent,
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedSnapshotProcessEvent) error {
+				if len(events) == 0 {
+					t.Fatalf("no events were captured")
+				}
+				for _, event := range events {
+					if event.Pid != event.Tid {
+						t.Fatalf("thread %d was captured", event.Tid)
+					}
+				}
+				return nil
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var processId int
+			timeout := 5 * time.Second
+			runner := utilstest.NewRunnerWithTest(t, testCase.runnerConfig)
+			var filterMap *ebpf.Map
+			if testCase.mntnsFilterMap != nil {
+				filterMap = testCase.mntnsFilterMap(runner.Info)
+			}
+
+			gdgt, err := gadgetapi.NewGadget(
+				"snapshot_process",
+				"latest",
+				timeout,
+				nil,
+				ExpectedSnapshotProcessEvent{},
+				filterMap,
+			)
+			require.NoError(t, err, "creating gadget")
+
+			gdgt.BeforeGadgetRun = func() error {
+				// Use the runner to generate an event
+				utilstest.RunWithRunner(t, runner, func() error {
+					pid, _ := testCase.generateEvent()
+					processId = pid
+					return nil
+				})
+				return nil
+			}
+
+			gdgt.StartGadget(t)
+			fd := processId
```

Comment:
nit: pass this variable directly to the next line to avoid an extra temp variable around.